conjure-runtime/src/per_host_clients.rs

```diff
@@ -0,0 +1,67 @@
+//! Per-host-client types.
+use std::collections::{hash_map, HashMap};
+
+use url::Url;
+
+/// A collection of clients for each host of a replicated service.
+pub struct PerHostClients<T> {
+    pub(crate) clients: HashMap<Host, T>,
+}
+
+impl<T> PartialEq for PerHostClients<T> {
```

Comment:
This impl is a bit jank - we need it for the Refreshable to work, but it's only correct when comparing PerHostClients for a single `per_host_clients` "lineage".

We could add some kind of disambiguator field but I don't think people will end up trying to compare instances in practice.